src/ray/object_manager/object_manager.cc

```diff
@@ -196,19 +207,48 @@ ray::Status ObjectManager::PullSendRequest(const ObjectID &object_id,
   return ray::Status::OK();
 }
 
-ray::Status ObjectManager::Push(const ObjectID &object_id, const ClientID &client_id,
-                                int retry) {
+void ObjectManager::HandlePushTaskTimeout(const ObjectID &object_id,
+                                          const ClientID &client_id) {
+  RAY_LOG(ERROR) << "Invalid Push request ObjectID: " << object_id
+                 << " after waiting for " << config_.push_timeout_ms << " ms.";
+  auto iter = unfulfilled_push_tasks_.find(object_id);
+  if (iter != unfulfilled_push_tasks_.end()) {
```

Comment:
Instead, can we do `RAY_CHECK(iter != unfulfilled_push_tasks_.end())`?

Reply:
Both should happen on the `main_service_`, so it shouldn't be an issue (@elibol can you confirm?). If they could happen concurrently then we would be in trouble because they both modify `unfulfilled_push_tasks_`.